### PR TITLE
Lock icon for latejoin GUI

### PR DESCRIPTION
**Pull request by Visne**
## About the PR 
Adds a lock icon to the latejoin GUI, when you don't have enough playtime for the role. This better distinguishes which roles you are not allowed to play, and which roles are full. The icon is borrowed from the action bar.

While I was there I also added a margin between the job icon and the job title, and I cleand up some of the code.

**Screenshots**
image

### CODE REVIEW

File: Content.Client/LateJoin/LateJoinGui.cs [hunk 1]
```diff
@@ -24,9 +22,12 @@ public sealed class LateJoinGui : DefaultWindow
         [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
         [Dependency] private readonly IClientConsoleHost _consoleHost = default!;
         [Dependency] private readonly IConfigurationManager _configManager = default!;
+        [Dependency] private readonly IEntitySystemManager _entitySystem = default!;
 
         public event Action<(EntityUid, string)> SelectedId;
 
+        private readonly SpriteSystem _sprites = IoCManager.Resolve<IEntitySystemManager>().GetEntitySystem<SpriteSystem>();
```
Comment (mirrorcult) [inline Content.Client/LateJoin/LateJoinGui.cs:29]: cache this properly in the ctor like gameticker is

File: Content.Client/LateJoin/LateJoinGui.cs [hunk 2]
```diff
@@ -226,35 +228,39 @@ private void RebuildUI()
...
 
                         var specifier = new SpriteSpecifier.Rsi(new ResourcePath("/Textures/Interface/Misc/job_icons.rsi"), prototype.Icon);
-                        icon.Texture = specifier.Frame0();
+                        icon.Texture = _sprites.Frame0(specifier);
                         jobSelector.AddChild(icon);
 
                         var jobLabel = new Label
                         {
-                            Text = value != null ?
+                            Text = " " + (value != null ?
```
Comment (mirrorcult) [inline Content.Client/LateJoin/LateJoinGui.cs:236]: is this done to have padding? it really shouldnt be done that way

Comment (Visne) [reply Content.Client/LateJoin/LateJoinGui.cs:236]: Why not? I find it much cleaner than a MinWidth Control, and I'm pretty sure it's done at some other places in the codebase.

File: Content.Client/LateJoin/LateJoinGui.cs (hunk 1 above)
Comment (Visne) [reply Content.Client/LateJoin/LateJoinGui.cs:29]: What do you mean with properly caching? And to me gameticker doesn't look like it is cached at all.

**Comment by Visne** [comment]
Ready for re-review

**Review by vulppine — APPROVED** [review]

**Pull request merged**